src/pextlib1.0/curl.c

```diff
@@ -59,12 +62,17 @@
 #define _CURL_MINIMUM_XFER_TIMEOUT	((long)(60))		/* 1 minute */
 #define _CURL_MINIMUM_PROGRESS_INTERVAL ((double)(0.2)) /* 0.2 seconds */
 
+/* pre-10.8 curl does not appear to support compression out of the box */
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_8
```

Comment:
This would be better done as a curl version check.

Reply:
Actually, it looks like curl supports compression back to at least 10.6. The use of https in the tests is likely the problem.

Reply:
Hmm, nope, `/usr/bin/curl --compressed https://www.whatsmyip.org/http-compression-test/` works fine, so it's not the TLS. But it isn't lack of CURLOPT_ENCODING either.

Reply:
My hunch was that the Apple curl wasn't compiled with zlib or something like that. If you have an old Mac around can you see what kind of request headers it's producing?  If that is the case it makes sense to just ban on OSX version.

Reply:
```
% /usr/bin/curl -v --compressed https://www.whatsmyip.org/http-compression-test/
* About to connect() to www.whatsmyip.org port 443 (#0)
*   Trying 204.11.35.98... connected
* Connected to www.whatsmyip.org (204.11.35.98) port 443 (#0)
* SSLv3, TLS handshake, Client hello (1):
* SSLv3, TLS handshake, Server hello (2):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS handshake, Server key exchange (12):
* SSLv3, TLS handshake, Server finished (14):
* SSLv3, TLS handshake, Client key exchange (16):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSL connection using DHE-RSA-AES256-SHA
* Server certificate:
* 	 subject: CN=random.whatsmyip.org
* 	 start date: 2019-12-24 19:43:20 GMT
* 	 expire date: 2020-03-23 19:43:20 GMT
* 	 subjectAltName: www.whatsmyip.org matched
* 	 issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
* 	 SSL certificate verify ok.
> GET /http-compression-test/ HTTP/1.1
> User-Agent: curl/7.19.7 (universal-apple-darwin10.0) libcurl/7.19.7 OpenSSL/0.9.8y zlib/1.2.3
> Host: www.whatsmyip.org
> Accept: */*
> Accept-Encoding: deflate, gzip
> 
< HTTP/1.1 200 OK
< Date: Fri, 31 Jan 2020 02:02:58 GMT
< Server: Apache
< X-Powered-By: Web Server Built Entirely of Legos
< Cache-Control: max-age=0
< Expires: Fri, 31 Jan 2020 02:02:58 GMT
< Vary: Accept-Encoding
< Content-Encoding: gzip
< Content-Length: 3177
< Content-Type: text/html; charset=UTF-8
< 
<!DOCTYPE html>
```
(actual page content trimmed)

Reply:
With current master (i.e. none of the changes in this PR applied), running both `curl fetch` and `curl fetch  --disable-compression` in Tcl sends the exact same headers in each case:
```
GET / HTTP/1.1
User-Agent: MacPorts/2.6.99 libcurl/7.19.7
Host: x.x.x.x:1234
Accept: */*
```

Reply:
The problem is that CURLOPT_ENCODING is not defined as a macro, but as part of an enum. Changing the #elif to an #else should fix it.